### ISSUE TITLE
Bootstrap 4 pagination page size dropdown not compatible?

### ISSUE THREAD

**Issue by SimonDKnight** [AllenFang/react-bootstrap-table#1130]
Hi, I recently upgraded my app to use bootstrap 4 and noticed the drop down for the page size does not work. I looked into the issue and noticed they have changed how the dropdown works in bootstrap 4 compared to 3.

So the main differences I can see is that in bootstrap 4 :

- The list items are just <a> tags and not <li> items
- When opened the containing div is given the class open in bootstrap 3 and show in bootstrap 4

Is there any chance of a fix for bootstrap 4?

Thanks again.

Simon

Oh here is an example of a dropdown in bootstrap 4

```<div class="dropdown">
  <button class="btn btn-secondary dropdown-toggle" type="button" id="dropdownMenuButton" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
    Dropdown button
  </button>
  <div class="dropdown-menu" aria-labelledby="dropdownMenuButton">
    <a class="dropdown-item" href="#">Action</a>
    <a class="dropdown-item" href="#">Another action</a>
    <a class="dropdown-item" href="#">Something else here</a>
  </div>
</div>


and here is how the dropdown works in bootstrap 3

<div class="btn-group">
  <button type="button" class="btn btn-default dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
    Action <span class="caret"></span>
  </button>
  <ul class="dropdown-menu">
    <li><a href="#">Action</a></li>
    <li><a href="#">Another action</a></li>
    <li><a href="#">Something else here</a></li>
    <li role="separator" class="divider"></li>
    <li><a href="#">Separated link</a></li>
  </ul>
</div>

**Comment by AllenFang** [comment]
I need to check this out and not very familiar with bootstrap 4, so not very sure that it can be compatible with 3 and 4.

Currently, Im pretty bust, so need some time to handle this issue, thanks. PR is welcome

**Comment by bar0191** [comment]
@SimonDKnight Did you ever find a fix for this? I am currently using bootstrap 4 in my app as well.

**Comment by JLoboRod** [comment]
There problem here basically is that boostrap4 uses a different class to show dropdown-menu. In bootstrap3 is **open** whereas in boostrap4 is **show**.
A workaround to make it work with both bootstrap 3 and 4 might be adding show as property as well as it has open and keep updating both.
In my case, I added some css to add that missing class for now.

**Comment by AllenFang** [comment]
@JLoboRod, sounds good if your PR is ready :)

**Comment by JLoboRod** [comment]
This should work for both bootstrap 3 and 4.

**Comment by AllenFang** [comment]
Released on `v4.0.0-beta.6`, The production release will be in this or next week, thanks your contribution.

**Comment by AllenFang** [comment]
Fixed on `v4.0.0`, after `v4.0.0`, `react-bootstrap-table` will compatible with bootstrap 3 and 4.
